package/base-files/files/sbin/opkg-keep

```diff
@@ -0,0 +1,24 @@
+#!/bin/sh
+for file in /etc/sysupgrade.conf /usr/lib/opkg/status /etc/rc.local; do
+    cp $file $file.bak
+    echo $file.bak >> /etc/sysupgrade.conf
+done
+
+cat <<\EOF > /etc/rc.local
+#!/bin/sh
+opkg update && {
+    opkg install diffutils
+    OPKGSTAT=/usr/lib/opkg/status
+    OPKGLIST=`diff $OPKGSTAT $OPKGSTAT.bak|grep '^> Package'|\
+              awk '{print $3}'|sort|xargs`
```

Comment:
As mentioned above, using diff is overkill here. Comparing two sorted package lists (once from .bak and once from the live file) should suffice